Signup

farhad-rezazadeh opened the pull request:
sign up and login feature with telegram user 
but /start command within running bot occur bug!!!

farhad-rezazadeh committed:
add button
add reply_keyboard to start.py

farhad-rezazadeh committed:
create sign_up.py
create new handler for handling sign up process
user can sign up with telegram account

farhad-rezazadeh committed:
create error.py
create function for different error situation

farhad-rezazadeh committed:
use error.py module in sign_up

farhad-rezazadeh committed:
add sign up handler to telegrambot

farhad-rezazadeh committed:
create login.py
create new handler for handling login process
user can log in with telegram account

farhad-rezazadeh committed:
add login handler to telegrambot

farhad-rezazadeh committed:
fix DosNotExist error and cancel message in login.py

farhad-rezazadeh committed:
remove error function for error.py

farhad-rezazadeh committed:
remover error function in login.py and sign_up.py

farhad-rezazadeh committed:
remove remaining dependencise

farhad-rezazadeh committed:
reformat code with black

YazdanRa commented on QBot/telegram_bot/telegrambot.py:
```diff
@@ -15,6 +17,8 @@ def main():
 
     handlers = [
         (start.HANDLER, 1),
+        (sign_up.HANDLER, 1),
```
it's better to use different groups for handlers

YazdanRa commented on QBot/telegram_bot/views/error.py:
```diff
@@ -0,0 +1,25 @@
+    context.bot.send_message(
+        chat_id,
+        text="cancel",
+        reply_markup=ReplyKeyboardMarkup(reply_keyboard, one_time_keyboard=True),
```
Using other keyboards in the cancel handler is not well.
besides, we should log the errors!

but the main idea is good 👍

YazdanRa replied:
If you need to find more search for `services` in the Django and `logger`

YazdanRa replied:
We don't need errors anymore we can remove them 👍 
we handle errors in `services.py` and we add it to the bot in `telegrambot.py` to find out more check that out =)

YazdanRa commented on QBot/telegram_bot/views/login.py:
```diff
@@ -0,0 +1,68 @@
+HANDLER = ConversationHandler(
+    entry_points=[MessageHandler(Filters.regex(r"(Login)"), logging_in)],
+    states={
+        BASE: [
```
For this part, we don't have a default login and a custom login.

the definition of login is to have a User that has the telegram field with specific details.

YazdanRa commented on QBot/telegram_bot/views/login.py:
```diff
@@ -0,0 +1,68 @@
+
+
+HANDLER = ConversationHandler(
+    entry_points=[MessageHandler(Filters.regex(r"(Login)"), logging_in)],
```
It's better to use regular text filters for specific "phrases" instead of regex. However, it's not wrong at all.

YazdanRa commented on QBot/telegram_bot/views/sign_up.py:
```diff
@@ -0,0 +1,81 @@
+        is_bot = update.message.from_user.is_bot
+        language_code = update.message.from_user.language_code
+
+        TelegramUser.objects.create(
```
SignUp will handle by the User model!

we will save all the Telegram account which are using the bot :)

Logged in User = a User object with a foreign-key to a TelegramUser!

YazdanRa reviewed (commented):
Shall I change the issues on my own?!
@farhad-rezazadeh

Commit:
Update register process

- rename "sign_up" to "register"
- fix register process
- fix group of handlers
- add signals file

YazdanRa reviewed (commented):
Check out the changes it will help you to figure out about the process

@farhad-rezazadeh 

feel free and ask about anything if it makes you uncomfortable 🙃

Commit:
Add global update processor and error handler

Commit:
Delete password message du to the security issues

Commit:
Add todo message for better logging

Sajjad-Arjmand committed:
test

Sajjad-Arjmand committed:
Update settings.py

deleted #test

YazdanRa commented on .vs/ProjectSettings.json:
```diff
@@ -0,0 +1,3 @@
+{
```
dear @Sajjad-Arjmand please add `.vs` directory to `.gitignore` these files should not be in git!

Sajjad-Arjmand replied:
ok. sorry this is my very first team project, but i'm a quick learner 😅

YazdanRa replied:
It's absolutely OK! 
feel free 🙃

YazdanRa commented:
The state of this issue will move automatically, and you don't need to change that on the board.

please DO NOT change cards on the board, and add (#ISSUE_NUMBER) at the end of your commit messages.

Sajjad-Arjmand committed:
Revert "test"

This reverts commit e97bd9e9e317c0b481ee5d9a15167dcb55fe4b97.
